crates/sui-types/src/gas_model/gas_v2.rs

```diff
@@ -314,10 +329,18 @@ impl SuiGasStatusAPI for SuiGasStatus {
 
     fn bucketize_computation(&mut self) -> Result<(), ExecutionError> {
         let gas_used = self.gas_status.gas_used_pre_gas_price();
-        let bucket_cost = get_bucket_cost(&self.cost_table.computation_bucket, gas_used);
-        // charge extra on top of `computation_cost` to make the total computation
-        // cost a bucket value
-        let gas_used = bucket_cost * self.gas_price;
+        let gas_used = if self.gas_status.gas_model_version > 5 {
+            if gas_used > 0 && gas_used % self.gas_rounding_step == 0 {
+                gas_used * self.gas_price
+            } else {
+                ((gas_used / self.gas_rounding_step) + 1) * self.gas_rounding_step * self.gas_price
```

Comment:
This may be a dumb question, but is there at all a risk of overflow here?

Reply:
I do not think so as we divide and multiply by the same value and then multiply by `gas_price`. In other words that feels as safe as the line above and what we do inside that function.
Having said so we are going to rework the gas model and we will be more conservative in those operations.
I do not think there is much to do right now

Reply:
we will be panicking on overflow which is the point I was making, is it not the case?
I think overall we made sure things are ok but we also need to review this code and build more safety around it.
The point I was making was that the line above is even more risky so to speak.
Thoughts?

Reply:
I think we are okay here. We do have a max gas budget of 50 SUI (txns over this limit will be rejected at signing) and `gas_used` cannot exceed this value.

Reply:
Is the fact that we don't sign transactions with huge budgets the thing that's protecting us here or are there other safe guards, because depending on where that check is done, it may not apply to a Dev Inspect transaction (which is not signed by validators in the first place).